src/main/java/org/sonar/plugins/stash/issue/MarkdownPrinter.java

```diff
@@ -52,55 +52,68 @@ public static String printSeverityMarkdown(String severity) {
   public static String printIssueNumberBySeverityMarkdown(List<Issue> report, String severity) {
     StringBuilder sb = new StringBuilder();
     long issueNumber = countIssuesBySeverity(report, severity);
-    sb.append("| ").append(severity).append(" | ").append(issueNumber).append(" |").append(NEW_LINE);
+    sb.append("| ").append(severity).append(" | ").append(issueNumber).append(" |")
+        .append(NEW_LINE);
 
     return sb.toString();
   }
 
-  public static String printIssueListBySeverityMarkdown(List<Issue> report, String sonarQubeURL, String severity) {
+  public String printIssueListBySeverityMarkdown(List<Issue> report, String severity) {
     StringBuilder sb = new StringBuilder();
 
     Map<String, Issue> rules = getUniqueRulesBySeverity(report, severity);
 
     // applying squid:S2864 optimization
     for (Map.Entry<String, Issue> rule : rules.entrySet()) {
       Issue issue = rule.getValue();
-      sb.append("| ").append(printIssueMarkdown(issue, sonarQubeURL)).append(" |").append(NEW_LINE);
+      sb.append("| ").append(printIssueMarkdown(issue)).append(" |").append(NEW_LINE);
     }
 
     return sb.toString();
   }
 
-  public static String printIssueMarkdown(Issue issue, String sonarQubeURL) {
+  public String printIssueMarkdown(Issue issue) {
     StringBuilder sb = new StringBuilder();
-    sb.append(MarkdownPrinter.printSeverityMarkdown(issue.severity())).append(issue.message())
-      .append(" [[").append(issue.ruleKey())
-      .append("]").append("(").append(sonarQubeURL).append("/")
-      .append(MarkdownPrinter.CODING_RULES_RULE_KEY).append(issue.ruleKey()).append(")]");
+    String message = issue.message();
+    String file = issuePathResolver.getIssuePath(issue);
+    if (file != null) {
+      String fileLink = link("`" + file + "`",
+          stashURL + "/projects/" + pr.project() +
+              "/repos/" + pr.repository() +
+              "/pull-requests/" + String.valueOf(pr.pullRequestId()) +
+              "/diff#" + file);
+      message = message.replace(file, fileLink);

```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "message" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)